sourcegraph/browser-extensions
# feat: support CXP extensions

🎉 **CXP extensions** 🎉

This adds support for CXP extensions and brings the browser extension to near feature parity with the Sourcegraph UI.

---

**Feature flag in the popup:**

![image](https://user-images.githubusercontent.com/1387653/43988656-86ffcc86-9cef-11e8-91cb-550a981c5cd8.png)

---

**Extensions page (shares UI components with https://sourcegraph.com/extensions):**

*TODO: don't show the option to add the extension to user/org/global settings*

![image](https://user-images.githubusercontent.com/1387653/43988690-ed021246-9cef-11e8-83a2-9e8ea6de964c.png)

---

**Line BG colors, annotations, and multiple CXP extensions:**

![2018-08-06 12 04 40](https://user-images.githubusercontent.com/1387653/43988801-e2e7b17e-9cf1-11e8-9f3a-bc44e8537474.gif)

---

**Setting a GitHub token for private repositories:**

*TODO: provide more visual feedback that the token was saved and is valid*

![2018-08-10 22 25 40](https://user-images.githubusercontent.com/1387653/43988811-1706d6a6-9cf2-11e8-8d10-24488ba9481d.gif)

Testing plan:

- `git checkout cxp-beta; git pull; npm run dev`
- Add/remove different extensions
- Switch between different Primary Sourcegraph URLs
- Enable/disable extensions
- Make sure Codecov shows up on https://github.com/ChacaS0/tempest/blob/master/cmd/purge.go
- Toggle the feature flag off and make sure existing functionality still works

**Known issues**

- CXP does not work on multiple files at a time (e.g. diff pages)

I have tested on:


cc @sqs
